actioncable/app/javascript/action_cable/index.js

```diff
@@ -0,0 +1,61 @@
+import Connection from "./connection"
+import ConnectionMonitor from "./connection_monitor"
+import Consumer from "./consumer"
+import INTERNAL from "./internal"
+import Subscription from "./subscription"
+import Subscriptions from "./subscriptions"
+
+const ActionCable = {
+  WebSocket: window.WebSocket,
+  logger: window.console,
+
+  createConsumer(url) {
+    if (url == null) {
+      const urlConfig = this.getConfig("url")
+      url = (urlConfig ? urlConfig : this.INTERNAL.default_mount_path)
+    }
+    return new ActionCable.Consumer(this.createWebSocketURL(url))
+  },
+
+  getConfig(name) {
+    const element = document.head.querySelector(`meta[name='action-cable-${name}']`)
+    return (element ? element.getAttribute("content") : undefined)
+  },
+
+  createWebSocketURL(url) {
+    if (url && !/^wss?:/i.test(url)) {
+      const a = document.createElement("a")
+      a.href = url
+      // Fix populating Location properties in IE. Otherwise, protocol will be blank.
+      a.href = a.href
+      a.protocol = a.protocol.replace("http", "ws")
+      return a.href
+    } else {
+      return url
+    }
+  },
+
+  startDebugging() {
+    this.debugging = true
+  },
+
+  stopDebugging() {
+    this.debugging = null
+  },
+
+  log(...messages) {
+    if (this.debugging) {
+      messages.push(Date.now())
+      this.logger.log("[ActionCable]", ...messages)
+    }
+  }
```

Comment:
Apologies if I wasn't clear in https://github.com/rails/rails/pull/34177#discussion_r225986157. I'm +1 on extracting a separate `logger` module to eliminate the circular dependencies. The interface here can just delegate to it, like I sketched in https://github.com/rails/rails/pull/34177#issuecomment-429460253.

Reply:
I did extract a separate logger module and eliminate the circular dependencies, but that's in https://github.com/rails/rails/pull/34258. Can you take a look at that PR? It's not exactly what you sketched but it's very similar. The main difference is that it enables applications to import exactly the ActionCable methods they need which offers the best tree shaking we can provide, while still keeping the index.js entry point clean and minimal.

Reply:
That PR looks very nice! I don't see any reason to keep those changes separate now. Would you prefer to close this PR and work from that one?

Reply:
Great! I think I'll just pull those commits into this branch and close that PR, since this one has most of the historical discussion.